service/storage_service.cc

```diff
@@ -485,8 +486,9 @@ future<> storage_service::topology_state_load() {
                 if (rs.ring.has_value()) {
                     if (!is_me(ip)) {
                         // Save ip -> id mapping in peers table because we need it on restart, but do not save tokens until owned
-                        co_await _sys_ks.local().update_tokens(ip, {});
```

Comment:
we lost `co_await _sys_ks.local().update_tokens(ip, {});` here. I'm not sure if it's important.

Reply:
if no tokens are stored yet' then storing an empty set is equivalent (yet I think it's better to store no tokens than to store an empty set).
If there were tokens than storing an empty set here is definitely wrong.  The mode should be decommissioned or removed to lose its tokens.